### PR TITLE
Add operator version & ArgoCD annotations to CRDs

### LINKED ISSUE #410: Support ArgoCD for the CRDs
When deploying Solr-Operator in [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) using its Helm chart, you may get this error message:
> reason: CustomResourceDefinition.apiextensions.k8s.io "solrclouds.solr.apache.org" is invalid: metadata.annotations: Too long: must have at most 262144 bytes

This is since ArgoCD uses `kubectl apply` by default, which then overflows some buffer. Explanation here: https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#replace-resource-instead-of-applying-changes

The same does not happen during normal `helm install` since Helm uses another method of installing CRDs.

This is not a bug in solr-operator, but there seems to be a very simple workaround, by adding this annotation to all our CRDs:

```yaml
metadata:
  annotations:
    argocd.argoproj.io/sync-options: Replace=true
```

This will tell Argo to use `kubectl replace` instead of apply, circumventing this limitation.

### PR DESCRIPTION
**Pull request by janhoy**
Add an annotation to our CRDs that will make ArgoCD replace instead of apply.

I have edited the three base CRDs, and also the dependency ZK CRD, but that is fragile, since the ZK crd may be overwritten on next upgrade...

### CODE REVIEW

**Comment by HoustonPutman** [comment]
As I mentioned in the issue, the Solr CRDs are auto-generated by controller-gen, so we need to go through and add the annotations after the generation. (Until controller-gen supports adding CRD annotations)

Will look into how this could work

**Comment by HoustonPutman** [comment]
@janhoy hacked together something that should work for our uses. I also included the version of the operator that the CRDs were made for, which will let users debug and make sure they have the correct CRD installed.

Also I did it in a way that it's done for the helm chart CRD file as well, since users using that are likely to have un-upgraded CRDs installed, so they need the version annotation more than anyone else.

Also included a small fix for the release wizard that I had locally

**Review by thelabdude — APPROVED** [review]
very cool solution!

File: hack/config/add_crds_annotations.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+files=("${CONFIG_DIRECTORY:-config}"/crd/bases/*)
+
+# Copy and package CRDs
+for file in "${files[@]}"; do
+  {
+    cat "${file}" | sed -e "/^    controller-gen.kubebuilder.io.version.*/a \\
+    operator.solr.apache.org\\/version: ${VERSION}\\
+    argocd.argoproj.io\\/sync-options: Replace=true"
+  } > "${file}.tmp" && mv "${file}.tmp" "${file}"
```
Comment (janhoy) [inline hack/config/add_crds_annotations.sh:36]: `sed -i` would do it in-place but may not work on some OSes

**Comment by janhoy** [comment]
Looks good, feel free to take over the PR and merge when done

File: hack/config/add_crds_annotations.sh (hunk 1 above)
Comment (HoustonPutman) [reply hack/config/add_crds_annotations.sh:36]: Yeah, `sed` differences between OSX and linux drive me insane, especially the `-i`, so it's not used anywhere in the solr-operator scripts. This works for now so I'm happy enough with it, we can always improve it in the future 🙂

**Comment by HoustonPutman** [comment]
The Zookeeper Operator CRD annotations won't be overwritten anymore either, and will be included in the release-artifacts.

**Pull request merged**